File: Training/1.2 Beginning/1.2.3-friday.py
'''
ID:   andy1li
LANG: PYTHON3
TASK: friday
'''

def is_leap(year): 
    return year%4==0 and (year%100 != 0 or year%400 == 0)

def days_in_month(year, month):
    length = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    return length[month] + (month == 1 and is_leap(year))

def solve():
    n = int(input())
    count = [0] * 7
    day_of_week = 0 # day of week: January 13, 1900 was a Saturday = 0 
    for year in range(1900, 1900+n):
        for month in range(12):
            count[day_of_week] += 1
            day_of_week += days_in_month(year, month)
            day_of_week %= 7

    print_line(' '.join(str(count[day]) for day in range(7)))

#------------------------------------------------------------------------------#

def print_line(line): print(line); f_out.write( f'{line}\n' )

task = 'friday'
with open(task+'.in') as f_in, open(task+'.out', 'w') as f_out:
    input = f_in.readline
    solve()

    

File: Training/1.2 Warm up/1.2.3-friday.cpp
/*
ID:   andy1li
LANG: C++
TASK: friday
*/

#include <bits/stdc++.h>
using namespace std;
using vi = vector<int>;

#define range(i, start, stop) for (size_t i=start; i<stop; ++i)
#define FOR(stop) for (size_t i=0; i<stop; ++i)
#define print_list(xs) FOR(xs.size()) cout<<xs[i]<<((i<xs.size()-1)?' ':'\n');

/*----------------------------------------------------------------------------*/

int is_leap(int year) {
    return year%4==0 && (year%100 != 0 || year%400 == 0);
}

int days_in_month(int year, int month) {
    vi length = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
    return length[month] + (month == 1 && is_leap(year));
}

void solve() {
    int n, day_of_week = 0; // day of week: January 13, 1900 was a Saturday = 0
    cin >> n; vi count(7, 0);

    range(year, 1900, 1900+n) {
        range(month, 0, 12) {
            count[day_of_week %= 7]++;
            day_of_week += days_in_month(year, month);
        }
    }
    print_list(count)
}

/*----------------------------------------------------------------------------*/

int main() {
    freopen("friday.in",  "r", stdin);
    freopen("friday.out", "w", stdout);
    solve();
}